### ISSUE TITLE
explanation with example how to install into HA

### ISSUE THREAD

**Issue by PE1NDO** [kvj/hass_Mtastic_MQTT#4]
Can you please give an example how to setup and install your integration?
I am far to stupid doing this with the few descriptive lines of text.
I know it's a lot of work, but I bet you there are more people like me around who would like to integrate Meshtastic into HA and they can't because of not beiing a python/HA guru.

**Comment by CraigBell** [comment]
Did you ever get this to work, I'm struggling with it too! I used MQTT Explorer on my local machine and identified the correct topics, but when it comes to HA, I have no clue how to get this to work!

**Comment by kvj** [comment]
Hey,

Do you guys know how to run HACS and install custom components via it?

**Comment by CraigBell** [comment]
Hi,

Yes, I got that far, installed, and used MQTT to validate the topics for my node. The bit I'm unclear on is, do I need to edit my mosquito config, or does the integration work directly with the meshtastic mqtt server?

This is the data that I entered for my node in the integration:

Node: !da5473**
Topic: msh/eu_868/2/e/LongFast/!da5473**
Key: AQ==
Stat topic: msh/eu_868/2/stat/!da5473**

**Comment by kvj** [comment]
> or does the integration work directly with the meshtastic mqtt server?

No, Home Assistant supports only single MQTT configuration. If you have your MQTT server configured in HA, you need to bridge selected topics

**Comment by CraigBell** [comment]
OK, then that is the bit that I'm missing then, makes total sense. In MQTT Explorer I see my nodes online and sending updates. 

Thank you, I'll look at bridging from your documentation, thank you.

**Comment by Coooogz** [comment]
I've think I've managed to bridge my MQTT server with meshtastic, however only seeing 2 random devices if I use MQTT Explorer onto my local HA mqtt server. I am yet to successfully connect and explore meshtastic mqtt using the explorer? Any tips?

**Comment by CraigBell** [comment]
For the Meshtatic server, I just entered the following:

Host: mqtt.meshtastic.org
Username meshdev 
Password I left blank and TLS was not selected.

Try that.

**Comment by Coooogz** [comment]
> For the Meshtatic server, I just entered the following:
> Host: mqtt.meshtastic.org Username meshdev Password I left blank and TLS was not selected.
> Try that.

Thanks Craig that got me on the right path and eventually managed to work out how to get the bridge working and get devices showing in HA. 

Still confused why I am seeing some (2 or 3) random devices under msh/ on HA when using the bridge. Also using devices in private channel, again they show up under msh/ rather than meshtastic/

**Comment by CraigBell** [comment]
Out of interest, how did you actually bridge in HA? I use Mosquito MQTT, but cannot figure out how to add the config, where is the file, did you use the editor or is it under configuration?

**Comment by kvj** [comment]
I suggest to all of you guys here to start sharing your configs (relevant parts) and/or screenshots. it's very challenging to understand where are you now and what are you trying to achieve.

I would personally avoid adding `mqtt.meshtastic.org` MQTT server to HA directly. It's a public server not designed to seve HA needs. But it's up to you

**Comment by Coooogz** [comment]
Using terminal, I created a file called /share/mosquitto/mosquitto.conf and then placed below in there.. (similar to what is stated on here) however added username. Password was not required. Monitor logs when you restart mosquitto and you should see it load config from there. 

_connection meshtastic
address mqtt.meshtastic.org
remote_username meshdev

topic +/+/!e2e2xxxx in 0 meshtastic/ msh/EU_868/2/
topic +/+/!3032yyyy in 0 meshtastic/ msh/EU_868/2/

topic !3032xxxx in 0 meshtastic/stat/ msh/EU_868/2/stat/
topic !e2e2yyyy in 0 meshtastic/stat/ msh/EU_868/2/stat/_

MQTT explorer was very helpful in confirming my nodes data was on both mesh MQTT and also HA mosquitto. 
I still have a few issues I am trying to work out though. 

Hope that helps. 

edit: the other change was to set in mosquito config page: 

active: true
folder: mosquitto

**Comment by CraigBell** [comment]
> Using terminal, I created a file called /share/mosquitto/mosquitto.conf and then placed below in there.. (similar to what is stated on here) however added username. Password was not required. Monitor logs when you restart mosquitto and you should see it load config from there.
> 
> _connection meshtastic address mqtt.meshtastic.org remote_username meshdev
> 
> topic +/+/!e2e2xxxx in 0 meshtastic/ msh/EU_868/2/ topic +/+/!3032yyyy in 0 meshtastic/ msh/EU_868/2/
> 
> topic !3032xxxx in 0 meshtastic/stat/ msh/EU_868/2/stat/ topic !e2e2yyyy in 0 meshtastic/stat/ msh/EU_868/2/stat/_
> 
> MQTT explorer was very helpful in confirming my nodes data was on both mesh MQTT and also HA mosquitto. I still have a few issues I am trying to work out though.
> 
> Hope that helps.
> 
> edit: the other change was to set in mosquito config page:
> 
> active: true folder: mosquito

Thanks, I finally picked this up again, I didn't have a mosquito folder, so I created one and used nano to add the data as per the documented example and yours too. I also changed the mosquito config to true and the folder was already there.

My last question, in the Meshtastic config for each node, what topics did you put there? I put msh/eu_868/2/e/LongFast/!daxxxx is that not correct?

**Comment by CraigBell** [comment]
> > Using terminal, I created a file called /share/mosquitto/mosquitto.conf and then placed below in there.. (similar to what is stated on here) however added username. Password was not required. Monitor logs when you restart mosquitto and you should see it load config from there.
> > _connection meshtastic address mqtt.meshtastic.org remote_username meshdev
> > topic +/+/!e2e2xxxx in 0 meshtastic/ msh/EU_868/2/ topic +/+/!3032yyyy in 0 meshtastic/ msh/EU_868/2/
> > topic !3032xxxx in 0 meshtastic/stat/ msh/EU_868/2/stat/ topic !e2e2yyyy in 0 meshtastic/stat/ msh/EU_868/2/stat/_
> > MQTT explorer was very helpful in confirming my nodes data was on both mesh MQTT and also HA mosquitto. I still have a few issues I am trying to work out though.
> > Hope that helps.
> > edit: the other change was to set in mosquito config page:
> > active: true folder: mosquito
> 
> Thanks, I finally picked this up again, I didn't have a mosquito folder, so I created one and used nano to add the data as per the documented example and yours too. I also changed the mosquito config to true and the folder was already there.
> 
> My last question, in the Meshtastic config for each node, what topics did you put there? I put msh/eu_868/2/e/LongFast/!daxxxx is that not correct?

I figured it out, it's all working now.

**Comment by Blackbeardaa** [comment]
I feel like I'm missing a step that is super obvious, and I'm just being big dumb. I added the repo and downloaded it on hacs. restarted HA. mosquito is running on HA. What's the nest step? where do configs get entered?

**Comment by Blackbeardaa** [comment]
I was right had the big stupid, didn't ever think to look in devices duh lol  thanks
